ONSdigital/dp-dataset-api
# Complete auditting process in API

### What

Audit process covers all endpoints, including check wrappers on routes and authentication wrapper. So even when a user is unauthenticated this will be auditted.

Also includes a tidy up, to align packages and handlers

### How to review

Check changes make logical sense and renamed variables and functions are sane. Unit tests pass as well as acceptance tests.

For acceptance tests to pass:
1) Make sure mongodb, neo4j, kafka and zookeeper are running
2) Run `dp-auth-api` with: `make debug`
3) Run `dp-dataset-api` with the following command: `make acceptance-publishing`
4) Go to `dp-api-tests` and checkout `feature/dataset-api-update-error-messages` branch
5) Make sure healthcheck has occurred on your running `dp-dataset-api` instance
6) In `dp-api-tests` run the following command: `go test publishing/datasetAPI/.`
7) Stop `dp-dataset-api` instance and run it again in web mode: `make acceptance-web`
8) Wait for healthcheck to occur before running the following in `dp-api-tests`:
`go test web/datasetAPI/.`

Both test runs should pass

See `dp-api-tests` pull request [here](https://github.com/ONSdigital/dp-api-tests/pull/100)

### Who can review

Anyone
